components/brave_referrals/browser/brave_referrals_service.h

```diff
@@ -33,7 +35,33 @@ class SimpleURLLoader;
 
 namespace brave {
 
-std::string GetAPIKey();
+class BraveReferralsHeaders {
+ public:
+  static BraveReferralsHeaders* GetInstance() {
```

Comment:
there is really no need to write the singleton boilerplate. Just a function that internally uses a static vector (or even better, flat_map) with hardcoded values would be enough.